tests/crc_test.c

```diff
@@ -0,0 +1,154 @@
+        .len = sizeof(DATA_32_VALUES),
+        .ptr = mut_DATA_32_VALUES,
+    };
+    const struct aws_byte_cursor input2 = {
+        .len = sizeof(TEST_VECTOR),
+        .ptr = mut_TEST_VECTOR,
+    };
+    res |= s_test_known_crc(func_name, func, "DATA_32_ZEROS", &input, KNOWN_CRC32_32_ZEROES, allocator);
+    res |= s_test_known_crc(func_name, func, "DATA_32_VALUES", &input1, KNOWN_CRC32_32_VALUES, allocator);
+    res |= s_test_known_crc(func_name, func, "TEST_VECTOR", &input2, KNOWN_CRC32_TEST_VECTOR, allocator);
+    return res;
+}
+
+/* helper function that groups crc32c tests*/
+static int s_test_known_crc32c(const char *func_name, crc_fn *func, struct aws_allocator *allocator) {
+    int res = 0;
+    uint8_t *mut_DATA_32_ZEROS = (uint8_t *)DATA_32_ZEROS;
+    uint8_t *mut_DATA_32_VALUES = (uint8_t *)DATA_32_VALUES;
+    uint8_t *mut_TEST_VECTOR = (uint8_t *)TEST_VECTOR;
+    const struct aws_byte_cursor input = {
+        .len = sizeof(DATA_32_ZEROS),
+        .ptr = mut_DATA_32_ZEROS,
+    };
+    const struct aws_byte_cursor input1 = {
+        .len = sizeof(DATA_32_VALUES),
+        .ptr = mut_DATA_32_VALUES,
+    };
+    const struct aws_byte_cursor input2 = {
+        .len = sizeof(TEST_VECTOR),
+        .ptr = mut_TEST_VECTOR,
+    };
+    res |= s_test_known_crc(func_name, func, "DATA_32_ZEROS", &input, KNOWN_CRC32C_32_ZEROES, allocator);
+    res |= s_test_known_crc(func_name, func, "DATA_32_VALUES", &input1, KNOWN_CRC32C_32_VALUES, allocator);
+    res |= s_test_known_crc(func_name, func, "TEST_VECTOR", &input2, KNOWN_CRC32C_TEST_VECTOR, allocator);
+
+    /*this tests three things, first it tests the case where we aren't 8-byte aligned*/
+    /*second, it tests that reads aren't performed before start of buffer*/
+    /*third, it tests that writes aren't performed after the end of the buffer.*/
+    /*if any of those things happen, then the checksum will be wrong and the assertion will fail */
+    uint8_t *s_non_mem_aligned_vector;
+    s_non_mem_aligned_vector = malloc(sizeof(DATA_32_VALUES) + 6);
+    memset(s_non_mem_aligned_vector, 1, sizeof(DATA_32_VALUES) + 6);
+    memcpy(s_non_mem_aligned_vector + 3, DATA_32_VALUES, sizeof(DATA_32_VALUES));
+
+    const struct aws_byte_cursor input3 = {
+        .len = sizeof(DATA_32_VALUES),
+        .ptr = s_non_mem_aligned_vector + 3,
+    };
+    res |= s_test_known_crc(func_name, func, "non_mem_aligned_vector", &input3, KNOWN_CRC32C_32_VALUES, allocator);
+    free(s_non_mem_aligned_vector);
+    return res;
+}
+
+/**
+ * Quick sanity check of some known CRC values for known input.
+ * The reference functions are included in these tests to verify that they aren't obviously broken.
+ */
+static int s_test_crc32c(struct aws_allocator *allocator, void *ctx) {
+    (void)ctx;
+
+    int res = 0;
+
+    res |= s_test_known_crc32c("aws_crc32c_new", aws_crc32c_new, allocator);
+
+    return res;
+}
+AWS_TEST_CASE(test_crc32c, s_test_crc32c)
```

Comment:
trivial: Maybe rename this test case to something like `crc32c_test` to be more consistent to the reset of tests in this repo?